file.go

```diff
@@ -79,13 +79,14 @@ func OpenFile(r io.ReaderAt, size int64, options ...FileOption) (*File, error) {
 			return nil, fmt.Errorf("reading page index of parquet file: %w", err)
 		}
 	}
-
-	if f.root, err = openColumns(f); err != nil {
+	var totalColumns int
+	f.root, totalColumns, err = openColumns(f)
+	if err != nil {
 		return nil, fmt.Errorf("opening columns of parquet file: %w", err)
 	}
 
 	schema := NewSchema(f.root.Name(), f.root)
-	columns := make([]*Column, 0, MaxColumnIndex+1)
+	columns := make([]*Column, 0, totalColumns)
```

Comment:
Maybe an alternative approach could be the following:
```suggestion
	columns := make([]*Column, 0, numLeafColumnsOf(f.root))
```

Reply:
Yeah I though about going through it again, it has the advantages of clarity but opted for the other option to avoid recursion and another trip down the tree. I figured with complex schema it would perform better if you open a lot of file.

I'm not super strong on that opinion though so if you prefer clarity, I'll do the change.

Reply:
I would usually default to simplicity unless we have been able to measure the compute cost. I agree that intuitively the second pass might appear as "more work" but computers are not always behaving in intuitive ways (e.g. when the data resides in CPU caches the extra pass is likely to be just as efficient as doing it in one run).

Reply:
Sounds good will make the change